tritium-core/src/main/java/com/palantir/tritium/event/InstrumentationFilters.java

```diff
@@ -36,6 +37,16 @@ public boolean shouldInstrument(@Nonnull Object _instance, @Nonnull Method _meth
         public boolean shouldInstrument(@Nonnull Object _instance, @Nonnull Method _method, @Nonnull Object[] _args) {
             return false;
         }
+    },
+
+    /** Instrument public interface methods. */
+    INSTRUMENT_PUBLIC_INTERFACE_METHODS {
+        @Override
+        public boolean shouldInstrument(@Nonnull Object _instance, @Nonnull Method method, @Nonnull Object[] _args) {
+            return Modifier.isPublic(method.getModifiers())
+                    && method.getDeclaringClass().isInterface()
+                    && !Modifier.isStatic(method.getModifiers());
```

Comment:
I would have expected this behavior to be identical to `INSTRUMENT_ALL`, would it make more sense to update INSTRUMENT_ALL instead? I don't think the dynamic proxy codepath will ever give us private or static methods, the byte-buddy implementation may need to be updated, but I might consider that an implementation detail of the byte-buddy instrumentation. What do you think?